In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd

In [ ]:
titanic = '/content/drive/MyDrive/Coding Dojo/Raw Data/titanic.csv'
df = pd.read_csv(titanic)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [ ]:
# 1 display the head of the dataframe
df.head()

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S
3            4         1       1  ...  53.1000  C123         S
4            5         0       3  ...   8.0500   NaN         S

[5 rows x 12 columns]

In [ ]:
# 2 what is the percentage of people who survived?
(df['Survived'].mean()) * 100

38.38383838383838

38.38% of people survived the sinking of the Titanic.

In [ ]:
# 3 How many of each sex survived?
df.groupby(['Sex'])[['Survived']].sum()

Survived
Sex             
female       233
male         109

In [ ]:
# 4 considering only the group of passengers who paid less than $10,
# what percentage of this group survived?
cost_filter = df['Fare'] < 10
cost_df = df.loc[cost_filter,:]
cost_df.head()

PassengerId  Survived  Pclass  ...    Fare Cabin  Embarked
0             1         0       3  ...  7.2500   NaN         S
2             3         1       3  ...  7.9250   NaN         S
4             5         0       3  ...  8.0500   NaN         S
5             6         0       3  ...  8.4583   NaN         Q
12           13         0       3  ...  8.0500   NaN         S

[5 rows x 12 columns]

In [ ]:
# 4 cont'dusing filtered DataFrame to find % of passengers who paid less than 
# $10 that survived
(cost_df['Survived'].mean()) * 100

19.940476190476193

19.94% of people who paid less than $10 survived the sinking of the Titanic.

In [ ]:
# 5 Overall, what is the average age of those who didn't survive?
# 6 Overall, what is the average age of those who did survive?
df.groupby(['Survived'])[['Age']].mean()

Age
Survived           
0         30.626179
1         28.343690

The average age of those who did not survive the sinking of the Titanic was 30.63 years old.
The average age of those who did survive the sinking of the Titanic was 28.34 years old.

In [ ]:
# 7 what is the average age of those who did and did not survive grouped by sex?
df.groupby(['Sex' , 'Survived'])[['Age']].mean()

Age
Sex                Survived           
(f, e, m, a, l, e) 0         25.046875
                   1         28.847716
(m, a, l, e)       0         31.618056
                   1         27.276022

The average age of females that did not survive the sinking of the Titanic was 25.05 years old. 
The average age of females that did survive the sinking of the Titanic was 28.85 years old.
The average age of males that did not survive the sinking of the Titanic was 31.62 years old.
The average age of males that did survive the sinking of the Titanic was 27.28 years old.